scarb/src/flock.rs

```diff
@@ -100,6 +100,12 @@ impl AsyncFileLockGuard {
         self.lock_kind
     }
 
+    pub fn into_file(mut self) -> tokio::fs::File {
+        self.file
+            .take()
+            .expect("failed to take ownership of the file")
+    }
+
```

Comment:
this is illegal. you will forget to call `unlock` on the file, which is done in `Drop` for (Async)FileLockGuard and is the exact purpose of existence of these types.

I think what you should do is to open the file as a stream and then make a `Body` out of it with https://docs.rs/reqwest/latest/reqwest/struct.Body.html#method.wrap_stream

Reply:
tried to do this after your previous comment, but was unable to, as the stream needs an ownership of the file (and lock), and all I could do was a reference. I also tried to do a ReadStream out of it, but no dice :( 
Perhaps I did something wrong, I do not exclude this possibility, but this solution sadly did not work for windows :(

Reply:
hmm, maybe you could use https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.try_clone then? 

Under the hood, this boils down to call to `fcntl` (`man fcntl`) with `F_DUPFD_CLOEXEC` on Unix and [`DuplicateHandle`](https://learn.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-duplicatehandle) on Windows. From quickly scanning docs it appears to me that this can only fail when we'll exceed the open file descriptor limit in the process, which is very unlikely.

Reply:
Hmm, interesting, it creates a new instance 🤔 seems to be working on windows as well, thanks :+1: https://github.com/THenry14/scarb/actions/runs/10740797646/job/29789733604
will read into this a bit more to better understand what happens :+1: Will of course make a change shortly

Reply:
Done in https://github.com/software-mansion/scarb/pull/1570/commits/687ab6004fec9cc8b82acfa6fe3d889f043678ad